tests/service-worker/test.js

```diff
@@ -1,36 +1,135 @@
 describe('tests', function() {
   'use strict';
 
-  var cacheName = 'testCache';
+  var storageKey = 'testCache';
+  var fakeEvent;
 
-  // Needed to prevent SW from bombing on what needs to be replaced
-  self.$urls = {
-    'http://localhost/wp-content/themes/my-theme/script.js': '32942374293',
-    'http://localhost/wp-content/themes/my-theme/style.css': '997898326'
-  };
-  self.$debug = 1;
+  beforeEach(function(done) {
+    // Needed to prevent SW from bombing on what needs to be replaced
+    self.$urls = {
+      'http://localhost:9876/socket.io/socket.io.js': '328947234',
+      'http://localhost:9876/karma.js': '32897324923'
+    };
+    self.$debug = 1;
 
-  beforeEach(function() {
     importScripts('/base/wp-sw-cache/lib/service-worker.js');
+
+    // Override our utility with a different storage object so we can manipulate
+    wpSwCache.storage = localforage.createInstance({ name: storageKey });
+    wpSwCache.cacheName = storageKey;
+
+
+    // A fake event for our fetches
+    fakeEvent = {
+      request: {},
+      respondWith: sinon.stub(),
+      waitUntil: sinon.stub()
+    };
+
+    // Clean up cache and storage before each test
+    wpSwCache.storage.clear().then(function() {
+      self.caches.delete(wpSwCache.cacheName).then(function(){
+        done();
+      });
+    });
   });
 
   it('This is a stub test', function() {
-    assert.strictEqual(true, true);
+    assert.isTrue(true);
   });
 
-  it('URLs which should be added to cache actually end up in cache and storage', function() {
-    // Simulates a basic install
-    assert.strictEqual(true, true);
+  it('URLs which should be added to cache actually end up in cache and storage', function(done) {
+    this.timeout(10000);
+
+    // Simulates a basic install -- check cache and localforage
+    wpSwCache.update().then(function() {
+      return self.caches.open(wpSwCache.cacheName).then(function(cache) {
+        return cache.keys().then(function(keys) {
+          assert.isTrue(keys.length == 2);
+
+          var cacheMatches = 0;
+          var storageMatches = 0;
+          return Promise.all(Object.keys(wpSwCache.urls).map(function(key) {
+            return cache.match(key).then(function(result) {
+              var isGood = (result && result.url === key);
+              assert.isTrue(isGood);
+              if(isGood) {
+                cacheMatches++;
+              }
+            })/*().then(function() {
```

Comment:
The storage check is somehow not matching.  I wonder if it's a context issue, somehow.